packages/substrate/src/base-transfer.ts

```diff
@@ -0,0 +1,74 @@
+import type { Config, Domain, Domainlike, SubstrateResource } from '@buildwithsygma/core';
+import { Bridge__factory } from '@buildwithsygma/sygma-contracts';
+import type { ExternalProvider } from '@ethersproject/providers';
+import { Web3Provider } from '@ethersproject/providers';
+import type { ApiPromise } from '@polkadot/api';
+import { constants, utils } from 'ethers';
+
+interface BaseTransferParams {
+  sourceDomain: Domainlike;
+  destinationDomain: Domainlike;
+  sourceNetworkProvider: ApiPromise;
+  resource: string | SubstrateResource;
+}
+
+export abstract class BaseTransfer {
+  sourceDomain: Domain;
+  destinationDomain: Domain;
+  sourceNetworkProvider: ApiPromise;
+  resource: SubstrateResource;
+  config: Config;
+
+  protected constructor(transfer: BaseTransferParams, config: Config) {
+    this.sourceDomain = config.getDomain(transfer.sourceDomain);
+    this.destinationDomain = config.getDomain(transfer.destinationDomain);
+    this.sourceNetworkProvider = transfer.sourceNetworkProvider;
+    const resources = config.getResources(this.sourceDomain);
+    const resource = resources.find(res => {
+      return typeof transfer.resource === 'string'
+        ? res.resourceId === transfer.resource
+        : res.resourceId === transfer.resource.resourceId;
+    });
+
+    if (resource) {
+      this.resource = resource as SubstrateResource;
+    } else {
+      throw new Error('Resource not found.');
+    }
+
+    this.config = config;
+  }
+
+  /**
+   * Method that checks whether the transfer
+   * is valid and route has been registered on
+   * the bridge
+   * @returns {boolean}
+   */
+  async isValidTransfer(): Promise<boolean> {
```

Comment:
The logic for `isValidTransfer` would be different when source is a substrate chain